internal/temporal/worker_deployment.go

```diff
@@ -0,0 +1,233 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the MIT License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2024 Datadog, Inc.
+
+package temporal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/serviceerror"
+	temporalClient "go.temporal.io/sdk/client"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	temporaliov1alpha1 "github.com/DataDog/temporal-worker-controller/api/v1alpha1"
+)
+
+// VersionStatus represents the status of a worker deployment version in Temporal
+type VersionStatus string
+
+const (
+	VersionStatusNotRegistered VersionStatus = "NotRegistered"
+	VersionStatusInactive      VersionStatus = "Inactive"
+	VersionStatusRamping       VersionStatus = "Ramping"
+	VersionStatusCurrent       VersionStatus = "Current"
+	VersionStatusDraining      VersionStatus = "Draining"
+	VersionStatusDrained       VersionStatus = "Drained"
+)
```

Comment:
I'm not super comfortable with all of the duplication with `v1alpha1` API. I imagine this will make refactoring take longer and create opportunities for bugs in mapping between types.

Why not use the `v1alpha1` types more directly? It seems like the business logic for querying k8s and temporal state could still be split out in the same way.

Reply:
Yep, makes sense. I've switched to use `v1alpha1` types where I can.